internal/payloads/reservation_payload.go

```diff
@@ -42,6 +42,10 @@ type GenericReservationResponsePayload struct {
 	Success *bool `json:"success" nullable:"true"`
 }
 
+type InstanceResponse struct {
+	*models.ReservationInstance
+}
```

Comment:
Nitpick: since this is named "instances" in JSON this could be also named "Instance".